apps/examples/ble_perfs/ble_perfs_main.c

```diff
@@ -0,0 +1,452 @@
+		.attr_handle = BLE_APP_HANDLE_CHAR_RMC_SYNC, .cb = ble_peri_cb_charact_rmc_sync, .arg = &g_perf_data},
+};
+
+static void ble_server_connected_cb(ble_conn_handle con_handle, ble_server_connection_type_e conn_type, uint8_t mac[BLE_BD_ADDR_MAX_LEN])
+{
+	char address_str[BLE_BD_ADDR_STR_LEN + 1] = {0,};
+
+	BLE_LOGD( "conn : %d / conn_type : %s(%d) \n", con_handle, get_conn_type_str(conn_type), conn_type);
+
+	addr_type_to_string(address_str, mac);
+	BLE_LOGD( "device address [%s] \n", address_str);
+
+	if (conn_type == BLE_SERVER_DISCONNECTED) {
+		perfs_ble_reset_data();
+	}
+
+	return;
+}
+
+static ble_server_init_config server_config = {
+	ble_server_connected_cb,
+	true,
+	gatt_profile, sizeof(gatt_profile) / sizeof(ble_server_gatt_t)};
+
+
+static void perfs_ble_usage(void)
+{
+	BLE_LOGD( "BLE Perfs Usage: \n\n" \
+		 " ble_perfs start \n" \
+		 " ble_perfs adv \n" \
+		 " ble_perfs stop \n" \
+		 " ble_perfs result \n");
+}
+
+static int perfs_ble_server_init(void)
+{
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	ble_result_e ble_ret = BLE_MANAGER_FAIL;
+
+	/*1. Enable WiFi
+	2. Enable BLE Manager
+	*/
+
+	res = wifi_manager_init(&g_wifi_callbacks);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		BLE_LOGD( "Fail to init wifi manager [%d] \n", res);
+		return -1;
+	}
+
+	usleep(100 * 1000);
+
+	ble_ret = ble_manager_init(&server_config);
+	if (ble_ret != BLE_MANAGER_SUCCESS) {
+		if (ble_ret == BLE_MANAGER_ALREADY_WORKING) {
+			BLE_LOGD( "ble init is already done \n");
+		} 
+		BLE_LOGD( "Fail to init ble manager \n");
+		return -1;
+	}
+
+	g_is_ble_serv_init = 1;
+	usleep(100 * 1000);
+
+	return 0;
+}
+
+static int perfs_ble_server_start(void)
+{
+	ble_data blue_data[1] = { 0, };
+	ble_result_e ret = BLE_MANAGER_FAIL;
+
+	if (g_is_ble_serv_init != 1) {
+		BLE_LOGD( "ble not initialised \n");
+		return -1;
+	}
+
+	/*3. Server Start = start advertising*/
+	blue_data->data = g_adv_raw;
+	blue_data->length = sizeof(g_adv_raw);
+
+	ret = ble_server_set_adv_data(blue_data);
+	if (ret != BLE_MANAGER_SUCCESS) {
+		BLE_LOGD( "Fail to set adv raw data ret:[%d]\n", ret);
+		return -1;
+	}
+	BLE_LOGD( "Set adv raw data ... ok\n");
+
+	blue_data->data = g_adv_resp;
+	blue_data->length = sizeof(g_adv_resp);
+
+	ret = ble_server_set_adv_resp(blue_data);
+	if (ret != BLE_MANAGER_SUCCESS) {
+		BLE_LOGD( "Fail to set adv resp data ret:[%d]\n", ret);
+		return -1;
+	}
+	BLE_LOGD( "Set adv resp data ... ok\n");
+
+	ret = ble_server_start_adv();
+	if (ret != BLE_MANAGER_SUCCESS) {
+		BLE_LOGD( "Fail to start adv ret:[%d]\n", ret);
+		return -1;
+	}
+	BLE_LOGD( "Start adv ... ok\n");
+
+	return 0;
+}
+
+static int perfs_ble_deinit(void)
+{
+	ble_result_e ret = BLE_MANAGER_FAIL;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+
+	res = wifi_manager_deinit();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		BLE_LOGD( "Fail to deinit wifi manager [%d] \n", res);
+		return -1;
```

Comment:
Is it okay not to deinit the ble manager in here?

Reply:
Do you mean initialize ?